salt/modules/selinux.py

```diff
@@ -604,3 +626,102 @@ def fcontext_apply_policy(name, recursive=False):
                 del new[key]
             ret['changes'].update({filespec: {'old': old, 'new': new}})
     return ret
+
+
+def port_get_policy(name, sel_type=None, protocol=None, port=None):
+    '''
+    .. versionadded:: Fluorine
+
+    Returns the current entry in the SELinux policy list as a
+    dictionary. Returns None if no exact match was found.
+
+    Returned keys are:
+
+    * sel_type (the selinux type)
+    * proto (the protocol)
+    * port (the port(s) and/or port range(s))
+
+    name
+        The protocol and port spec. Can be formatted as (tcp|udp)/(port|port-range).
+
+    sel_type
+        The SELinux Type.
+
+    protocol
+        The protocol for the port (tcp|udp). Required if name is not formatted.
+
+    port
+        The port or port range. Required if name is not formatted.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' selinux.port_get_policy tcp/80
+        salt '*' selinux.port_get_policy foobar protocol=tcp port=80
+    '''
+    (protocol, port) = _parse_protocol_port(name, protocol, port)
+    re_spacer = '[ ]+'
+    re_sel_type = sel_type if sel_type else r'\w+'
+    cmd_kwargs = {'spacer': re_spacer,
+                  'sel_type': re_sel_type,
+                  'protocol': protocol,
+                  'port': port, }
+    cmd = 'semanage port -l | egrep ' + \
+          "'^{sel_type}{spacer}{protocol}{spacer}((.*)*)[ ]{port}($|,)'".format(**cmd_kwargs)
+    port_policy = __salt__['cmd.shell'](cmd, ignore_retcode=True)
+    if port_policy == '':
+        return None
+
+    parts = re.match(r'^(\w+)[ ]+(\w+)[ ]+([\d\-, ]+)', port_policy)
+    return {
+        'sel_type': parts.group(1).strip(),
+        'protocol': parts.group(2).strip(),
+        'port': parts.group(3).strip(), }
+
+
+def port_add_or_delete_policy(action, name, sel_type=None, protocol=None, port=None, sel_range=None):
```

Comment:
This seems like a cumbersome name for a function, and also redundant since you're making the user specify the action as an argument. How about changing this function name to `_port_add_or_delete_policy()` and then making two separate functions to add and delete, each of which is a one-liner that calls the common function. For example:

```python
def port_add_policy(name, sel_type=None, protocol=None, port=None, sel_range=None):
    '''
    <docstring goes here>
    '''
    return _port_add_or_delete_policy('add', name, sel_type, protocol, port, sel_range)


def port_delete_policy(name, sel_type=None, protocol=None, port=None, sel_range=None):
    '''
    <docstring goes here>
    '''
    return _port_add_or_delete_policy('delete', name, sel_type, protocol, port, sel_range)
```

Reply:
This is the same naming scheme as the rest of the modules in this execution module.

https://github.com/saltstack/salt/blob/develop/salt/modules/selinux.py#L483

Probably good to rename all of them honestly, but i think that can be done in a different pr.

Reply:
Yes, I pretty much copied the fcontext naming/flow and altered it accordingly. I'll make a note to put together another PR - thanks for the feedback from both of you!

Reply:
I submitted PR #47622

Reply:
Awesome thanks @leeclemens